asynch/proto/connection.py

```diff
@@ -806,7 +806,9 @@ def substitute_params(query: str, params: Mapping[str, Any]) -> str:
         if not isinstance(params, Mapping):
             raise ValueError("Parameters are expected to be a mapping")
 
-        return query.format(**escape_params(params))
+        if query_is_format_style(query):
+            return query.format(**escape_params(params))
```

Comment:
Is it a good idea to throw warning message about "the .format support is experimental and may be removed in future"? Or just remove this support and remove the corresponding tests which would be:
- easy because it is not a big deal neither to remove the tests nor to support the introduced the feature;
- good for the maintenance's sake because people can discover and use this `format` feature which is not what SQLAlchemy supports.

Another pro for preserving the C-style and abandoning the `format` support is this [quickstart](https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html) page for the clickhouse-driver where I see no `{var}` style, but `%(var)spec`. So, @baconfield , maybe it is advisable to remove this feature as @dmkulazhenko proposes, but it is up to you, certainly.

Reply:
@baconfield , BTW, please add this PR in the CHANGELOG.